unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/FriendsController.cs

```diff
@@ -341,10 +375,19 @@ public async UniTask<IReadOnlyList<string>> GetFriendsAsync(string userNameOrId,
         public void GetFriendRequests(int sentLimit, int sentSkip, int receivedLimit, int receivedSkip) =>
             apiBridge.GetFriendRequests(sentLimit, sentSkip, receivedLimit, receivedSkip);
 
-        public async UniTask<List<FriendRequest>> GetFriendRequestsAsync(int sentLimit, int sentSkip, int receivedLimit, int receivedSkip, CancellationToken cancellationToken)
+        public async UniTask<IEnumerable<FriendRequest>> GetFriendRequestsAsync(int sentLimit, int sentSkip, int receivedLimit, int receivedSkip, CancellationToken cancellationToken)
         {
             cancellationToken.ThrowIfCancellationRequested();
 
+            if (useSocialApiBridge)
+            {
+                var sent = outgoingFriendRequestsByTimestamp.Values.Skip(sentSkip).Take(sentLimit);
+                var received = incomingFriendRequestsByTimestamp.Values.Skip(receivedSkip).Take(receivedLimit);
+                var result = received.Concat(sent);
+
+                return result;
+            }
+
```

Comment:
Replace LinQ, this allocates a lot of unneded memory.

something like this would be better: 
```csharp
if (useSocialApiBridge)
{
    int sentCount = outgoingFriendRequestsByTimestamp.Values.Count;
    int receivedCount = incomingFriendRequestsByTimestamp.Values.Count;

    List<FriendRequest> result = new List<FriendRequest>(sentCount - sentSkip + receivedCount - receivedSkip);

    int index = 0;
    foreach (var receivedRequest in incomingFriendRequestsByTimestamp.Values)
    {
        if (index >= receivedSkip && index < receivedSkip + receivedLimit)
        {
            result.Add(receivedRequest);
        }
        index++;
    }

    index = 0;
    foreach (var sentRequest in outgoingFriendRequestsByTimestamp.Values)
    {
        if (index >= sentSkip && index < sentSkip + sentLimit)
        {
            result.Add(sentRequest);
        }
        index++;
    }

    return result;
}
```

Review it, it was done by chatgpt and might have issues.